chains/runtime-common/src/relay.rs

```diff
@@ -810,6 +862,113 @@ where
 
         Ok(())
     }
+
+    fn process_xcm_local_container_eth_transfer(
+        eth_transfer_data: EthTransferData,
+    ) -> DispatchResult {
+        let (para_id, beneficiary, fee) = match eth_transfer_data.destination {
+            Destination::ForeignAccountId32 { para_id, id, fee } => (
+                para_id,
+                Location::new(0, [AccountId32 { network: None, id }]),
+                fee,
+            ),
+            Destination::ForeignAccountId20 { para_id, id, fee } => (
+                para_id,
+                Location::new(
+                    0,
+                    [AccountKey20 {
+                        network: None,
+                        key: id,
+                    }],
+                ),
+                fee,
+            ),
+            _ => {
+                log::error!(
+                    "EthTokensLocalProcessor: unsupported destination for container transfer: {:?}",
+                    eth_transfer_data.destination
+                );
+                return Ok(());
+            }
+        };
+
+        let container_location = Location::new(0, [Parachain(para_id)]);
+
+        let token_reanchored = match eth_transfer_data.token_location.reanchored(
+            &container_location,
+            &<T as pallet_xcm::Config>::UniversalLocation::get(),
+        ) {
+            Ok(loc) => loc,
+            Err(e) => {
+                log::error!(
+                    "EthTokensLocalProcessor: failed to reanchor token location: {:?}",
+                    e
+                );
+                return Ok(());
+            }
+        };
+
+        let container_location_reanchored = match container_location.clone().reanchored(
+            &container_location,
+            &<T as pallet_xcm::Config>::UniversalLocation::get(),
+        ) {
+            Ok(loc) => loc,
+            Err(e) => {
+                log::error!(
+                    "EthTokensLocalProcessor: failed to reanchor native container token location: {:?}",
+                    e
+                );
+                return Ok(());
+            }
+        };
+
+        let asset_fee: Asset = (container_location_reanchored.clone(), fee).into();
+        let asset_to_deposit: Asset = (token_reanchored.clone(), eth_transfer_data.amount).into();
+
+        let inbound_queue_pallet_index = InboundQueuePalletInstance::get();
+        let network = EthereumNetwork::get();
+
+        let remote_xcm = Xcm::<()>(vec![
+            DescendOrigin(PalletInstance(inbound_queue_pallet_index).into()),
+            UniversalOrigin(GlobalConsensus(network)),
+            WithdrawAsset(vec![asset_fee.clone()].into()),
+            BuyExecution {
+                fees: asset_fee.clone(),
```

Comment:
I believe we should take it from the inbound message, like we do it [here](https://github.com/paritytech/polkadot-sdk/blob/19320f104fc4b5cb6663cffc41f340b6b5239be8/bridges/snowbridge/primitives/inbound-queue/src/v1.rs#L336).